central/processlisteningonport/datastore/datastore_impl_test.go

```diff
@@ -318,84 +265,100 @@ func (suite *PLOPDataStoreTestSuite) TestPLOPAddOpenTwice() {
+				sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+				sac.ResourceScopeKeys(resources.Namespace),
+				sac.ClusterScopeKeys(fixtureconsts.Cluster1),
+			)),
+			expectAllowed: true,
 		},
-	})
-
-	// Verify that newly added PLOP object doesn't have Process field set in
-	// the serialized column (because all the info is stored in the referenced
-	// process indicator record)
-	newPlopsFromDB := suite.getPlopsFromDB()
-	suite.Len(newPlopsFromDB, 1)
-
-	expectedPlopStorage := &storage.ProcessListeningOnPortStorage{
-		Id:                 newPlopsFromDB[0].GetId(),
-		Port:               plopObjects[0].GetPort(),
-		Protocol:           plopObjects[0].GetProtocol(),
-		CloseTimestamp:     plopObjects[0].GetCloseTimestamp(),
-		ProcessIndicatorId: indicators[0].GetId(),
-		Closed:             false,
-		Process:            nil,
-		DeploymentId:       fixtureconsts.Deployment1,
-		PodUid:             fixtureconsts.PodUID1,
-		ClusterId:          fixtureconsts.Cluster1,
-		Namespace:          fixtureconsts.Namespace1,
 	}
 
-	suite.Equal(expectedPlopStorage, newPlopsFromDB[0])
-}
-
-// TestPLOPAddCloseTwice: Add the same closed PLOP twice
-// There should only be one PLOP in the storage
-func (suite *PLOPDataStoreTestSuite) TestPLOPAddCloseTwice() {
-	indicators := getIndicators()
-
-	plopObjects := []*storage.ProcessListeningOnPortFromSensor{&closedPlopObject}
-
-	// Prepare indicators for FK
-	suite.NoError(suite.indicatorDataStore.AddProcessIndicators(
-		suite.hasWriteCtx, indicators...))
-
-	// Add PLOP referencing those indicators
-	suite.NoError(suite.datastore.AddProcessListeningOnPort(
-		suite.hasWriteCtx, fixtureconsts.Cluster1, plopObjects...))
-
-	// Add the same PLOP again
-	suite.NoError(suite.datastore.AddProcessListeningOnPort(
-		suite.hasWriteCtx, fixtureconsts.Cluster1, plopObjects...))
-
-	// Fetch inserted PLOP back
-	newPlops, err := suite.datastore.GetProcessListeningOnPort(
-		suite.hasWriteCtx, fixtureconsts.Deployment1)
-	suite.NoError(err)
-
-	suite.Len(newPlops, 0)
+	for name, c := range cases {
+		suite.Run(name, func() {
+			newPlops, err := suite.datastore.GetProcessListeningOnPort(
+				c.ctx, fixtureconsts.Deployment1)
+			if c.expectAllowed {
+				suite.NoError(err)
+				suite.Len(newPlops, 1)
+				suite.Equal(newPlops[0], &storage.ProcessListeningOnPort{
+					ContainerName: "test_container1",
+					PodId:         fixtureconsts.PodName1,
+					PodUid:        fixtureconsts.PodUID1,
+					DeploymentId:  fixtureconsts.Deployment1,
+					ClusterId:     fixtureconsts.Cluster1,
+					Namespace:     fixtureconsts.Namespace1,
+					Endpoint: &storage.ProcessListeningOnPort_Endpoint{
+						Port:     1234,
+						Protocol: storage.L4Protocol_L4_PROTOCOL_TCP,
+					},
+					Signal: &storage.ProcessSignal{
+						Name:         "test_process1",
+						Args:         "test_arguments1",
+						ExecFilePath: "test_path1",
+					},
+				})
+			} else {
+				suite.ErrorIs(err, sac.ErrResourceAccessDenied)
```

Comment:
nit: I'd ensure here that no PLOP was returned in case of error.

Reply:
Done. Also not a nit.